packages/calcite-components/src/utils/focusTrapComponent.ts

```diff
@@ -73,7 +76,7 @@ export function activateFocusTrap(
   component: FocusTrapComponent,
   options?: Parameters<_FocusTrap["activate"]>[0],
 ): void {
-  if (!component.focusTrapDisabled) {
+  if ((component.focusTrapDisabledOverride && !component.focusTrapDisabledOverride()) || !component.focusTrapDisabled) {
```

Comment:
@Elijbet the or `||` operator will make it so that if `component.focusTrapDisabledOverride && !component.focusTrapDisabledOverride()` returns `false` it will use `!component.focusTrapDisabled`.

I think it should use the `??` operator here so that it doesn't do that.

I think this would be better:

```
if ((component.focusTrapDisabledOverride ? !component.focusTrapDisabledOverride()) : !component.focusTrapDisabled) {
```

Reply:
You're right, when focusTrapDisabledOverride is defined, the component should take over with its own logic.